### ISSUE TITLE
[stable/wordpress] cannot use an existing PVC

### ISSUE THREAD

**Issue by nmacherey** [helm/charts#21144]
**Describe the bug**

The existing PVC is attached and mounted but the content is erased event when `wordpressSkipInstall` is `true`

**Version of Helm and Kubernetes**:

Helm: 2.14.3
Kubernetes: 1.15.9-gke.8

**Which chart**:

stable/wordpress

**What happened**:

I am unable to use an existing PVC from a previous wordpress version (full wordpress code into it). The PVC content is emptied. 

**What you expected to happen**:

The installation relies on the existing disk content and does not replace anything.

**How to reproduce it** (as minimally and precisely as possible):

Here is the `values.yaml` used:

```yaml
wordpressUsername: <changeme>
wordpressPassword: <changeme>
wordpressEmail: change@me.com
wordpressFirstName: My
wordpressLastName: User
wordpressBlogName: My Blog
wordpressScheme: http
wordpressSkipInstall: true
allowOverrideNone: true
#customHTAccessCM: wordpress-htaccess
updateStrategy:
  type: Recreate
replicaCount: 1

mariadb:
  enabled: false

externalDatabase:
  host: changeme
  user: changeme
  password: changeme
  database: wordpress
  port: 3306

service:
  type: ClusterIP
  port: 80
  httpsPort: 443
  httpsTargetPort: https
  metricsPort: 9117
  annotations: {}

## Allow health checks to be pointed at the https port
healthcheckHttps: false

## Configure extra options for liveness and readiness probes
## ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#configure-probes)
livenessProbe:
  enabled: true
  initialDelaySeconds: 120
  periodSeconds: 10
  timeoutSeconds: 5
  failureThreshold: 6
  successThreshold: 1

readinessProbe:
  enabled: true
  initialDelaySeconds: 30
  periodSeconds: 10
  timeoutSeconds: 5
  failureThreshold: 6
  successThreshold: 1

ingress:
  enabled: false

persistence:
  enabled: true
  existingClaim: pv-claim-wordpress

## Configure resource requests and limits
## ref: http://kubernetes.io/docs/user-guide/compute-resources/
##
resources:
  requests:
    memory: 1024Mi
    cpu: 600m

metrics:
  enabled: false
````

The claim if you need it:

```yaml
apiVersion: v1
kind: PersistentVolume
metadata:
  name: pv-wordpress
  namespace: xmotion-wordpress
spec:
  storageClassName: ""
  capacity:
    storage: 50G
  accessModes:
    - ReadWriteOnce
  gcePersistentDisk:
    pdName: xm-wordpress-data
    fsType: ext4

---

apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: pv-claim-wordpress
  namespace: xmotion-wordpress
spec:
  # It's necessary to specify "" as the storageClassName
  # so that the default storage class won't be used, see
  # https://kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1
  storageClassName: ""
  volumeName: pv-wordpress
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 50G
```

**Anything else we need to know**:

The original disk do not come from an old bitnami wordpress container but the official one.

**Comment by juan131** [comment]
Hi @nmacherey 

As you can see at https://github.com/helm/charts/pull/21099 and https://github.com/helm/charts/issues/21098 WordPress has been moved to non-root and existing data from previous versions is not compatible anymore.

As can see in the README (see the link below), you'll need to migrate your data in order to upgrade to the latest version:

- https://github.com/helm/charts/tree/master/stable/wordpress#to-900

**Comment by nmacherey** [comment]
Hi Juan,

Thanks for your answer. I’ve migrated the full directory and it seams to work now. What I did is the following…


1. Create a VM on the cloud and attach the volume.
2. Login to the new VM instance and mount the volume at some path.
3. Move wp-config.php and wp-content folder to /path/to/my/volume/wordpress directory
4. Remove other files
5. I also had to edit manually the wp-config content to match with the new config parameters
6. Shut down the VM & remove it
7. Reattach the volume to the POD using the PVC

Regards
Nicolas
